modules/sagemaker/sagemaker-ground-truth-labeling/labeling_step_function/lambda/update_feature_store.py

```diff
@@ -0,0 +1,242 @@
+                if len(delete_entries) == CONCURRENT_PROCESSING_BATCH_SIZE:
+                    multi_threaded_batch_delete_message_from_sqs_and_save_to_feature_store(
+                        queue_url=queue_url,
+                        feature_group_name=feature_group_name,
+                        delete_entries=delete_entries,
+                        feature_store_records=feature_store_records,
+                    )
+                    delete_entries = []
+                    feature_store_records = {}
+
+    # Delete any remaining messages after for loop
+    if delete_entries:
+        multi_threaded_batch_delete_message_from_sqs_and_save_to_feature_store(
+            queue_url=queue_url,
+            feature_group_name=feature_group_name,
+            delete_entries=delete_entries,
+            feature_store_records=feature_store_records,
+        )
+
+    logger.info(f"Processed labeling job output, {len(record_source_to_receipt_handle)} remaining messages")
+
+    return record_source_to_receipt_handle
+
+
+def transform_labeling_result_to_record(
+    labeling_result: Dict[str, Any],
+    feature_group_definitions: Dict[str, List[Union[str, int]]],
+) -> List[Dict[str, str]]:
+    labeling_result_location = VERIFICATION_ATTRIBUTE_NAME if VERIFICATION_STEP else LABELING_ATTRIBUTE_NAME
+    if "failure-reason" in labeling_result[f"{labeling_result_location}-metadata"]:
+        return []
+
+    creation_date = datetime.datetime.strptime(
+        labeling_result[f"{LABELING_ATTRIBUTE_NAME}-metadata"]["creation-date"],
+        "%Y-%m-%dT%H:%M:%S.%f",
+    ).timestamp()
+
+    record = [
+        create_record_feature(SOURCE_KEY.replace("-", "_"), str(labeling_result[SOURCE_KEY])),
+        create_record_feature("event_time", str(int(round(creation_date)))),
+        create_record_feature(
+            "labeling_job",
+            str(labeling_result[f"{LABELING_ATTRIBUTE_NAME}-metadata"]["job-name"]),
+        ),
+    ]
+
+    if VERIFICATION_STEP:
+        status = (
+            APPROVED_STATUS
+            if labeling_result[f"{VERIFICATION_ATTRIBUTE_NAME}"] == VERIFICATION_APPROVED_CLASS
+            else REJECTED_STATUS
+        )
+
+        if status == REJECTED_STATUS:
+            return []
+        else:
+            record.append(create_record_feature("status", status))
+
+    for feature_name, output_key in feature_group_definitions.items():
+        # access nested value using key list
+        feature_value = reduce(operator.getitem, output_key, labeling_result)  # type: ignore
+        record.append(create_record_feature(feature_name, str(feature_value)))
+    return record
+
+
+def create_record_feature(feature_name: str, feature_value: str) -> Dict[str, str]:
+    return {"FeatureName": feature_name, "ValueAsString": feature_value}
+
+
+def multi_threaded_batch_delete_message_from_sqs_and_save_to_feature_store(
```

Comment:
Decided to multi-thread here due to sagemaker feature store not having a batch put operation. Running single threaded would cause issues at scale with lambda time limits.